### PR TITLE
Congratulatory Message for Pages Goal 137

### PR DESCRIPTION
**Pull request by tjrelyts**
## Summary of change

I added text that congratulates the user when they achieve their pages goal. It also relays progress percentages correctly.

## Additional context

image

## Related issue

#137

### CODE REVIEW

**Comment by knjk04** [comment]
Thanks for this. Before I review this pull request, could you fix the merge conflicts?

File: .gitignore [hunk 1]
```diff
@@ -39,3 +39,4 @@ build/
 
 ### VS Code ###
 .vscode/
+pnpm-lock.yaml
```
Comment (knjk04) [inline .gitignore:42]: Could you undo this change?

**Comment by knjk04** [comment]
There's a bug with resetting the congratulatory message. Once that's fixed and merged in, you can rebase/merge to add those commits to your branch

**Comment by tjrelyts** [comment]
> 
> 
> There's a bug with resetting the congratulatory message. Once that's fixed and merged in, you can rebase/merge to add those commits to your branch (issue #152, PR: #155)

Got it. Will look into it and fix it.

**Comment by tjrelyts** [comment]
Sorry for taking so long. Was waiting for that issue to be solved in PR #155. I have to go through the code and fix all the merge conflicts now with the updated code from #155. So as soon as I am done that, I should be able to submit.

**Comment by knjk04** [comment]
OK, thanks

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalForm.java [hunk 2]
```diff
@@ -140,7 +140,7 @@ private void configureBinder() {
 
         binder.forField(targetToRead)
               .asRequired("Please enter in a target goal")
-              .withValidator(target -> target >= 1, "Target must be at least 1")
+              .withValidator(target -> target > 0, "Target must be at least 1")
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalForm.java:143]: `>= 1` was fine. Could you revert this change?

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java [hunk 3]
```diff
@@ -146,12 +151,16 @@ private void updateReadingGoal(int targetToRead, ReadingGoal.GoalType goalType)
...
         if (goalType.equals(ReadingGoal.GoalType.BOOKS)) {
             readingGoal.setText(haveRead + howManyReadThisYear + outOf + + targetToRead + getPluralized(" book", targetToRead));
-            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear));
+            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear, goalType));
             booksToRead.setText(calculateBooksToRead(targetToRead, howManyReadThisYear));
+            togglePagesGoalInfo(false, hasReachedGoal);
             toggleBooksGoalInfo(true, hasReachedGoal);    // show book goal-specific information
         } else {
             readingGoal.setText(haveRead + howManyReadThisYear + outOf + targetToRead + getPluralized(" page", targetToRead));
+            goalProgress.setText(calculateProgress(targetToRead, howManyReadThisYear, goalType));
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java:160]: Since this line is a duplicate (it also appears in the `if` statement, this could be moved outside of the if/else

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java [hunk 4]
```diff
@@ -146,12 +151,16 @@ private void updateReadingGoal(int targetToRead, ReadingGoal.GoalType goalType)
...
+            booksToRead.setText(calculateBooksToRead(targetToRead, howManyReadThisYear));
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java:161]: You're using `booksToRead` here while the goal type is pages. If you still want to use that variable, I'd recommend renaming it and also considering whether you want to rename the similarly named `pagesToRead`

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java [hunk 5]
```diff
@@ -78,6 +78,11 @@
      */
     Span booksToRead;
 
+    /**
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java:81]: I know there's a javadoc comment for the other instance variables, but I was going to refactor that. Could you rename this variable to something descriptive such that including the javadoc comment would be redundant?

File: src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java [hunk 6]
```diff
@@ -213,25 +222,62 @@ private void toggleBooksGoalInfo(boolean isOn, boolean hasReachedGoal) {
...
-        LOGGER.log(Level.INFO, "Books still to read: " + booksStillToRead);
+        int stillToRead = toReadThisYear - readThisYear;
 
-        if (booksStillToRead <= 0) {
-            schedule = TARGET_MET;
-        } else {
-            int howManyBehindOrAhead = howFarAheadOrBehindSchedule(booksToReadThisYear, booksReadThisYear);
-            schedule = String.format("You are %d "+ getPluralized("book", howManyBehindOrAhead) + " %s schedule",
-                    howManyBehindOrAhead, behindOrAheadSchedule(booksReadThisYear, shouldHaveRead(booksToReadThisYear)));
+        if (goalType.equals(ReadingGoal.GoalType.BOOKS)) {
```
Comment (knjk04) [inline src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java:243]: Now that the progress is also being shown for a pages goal, this `if` block looks very similar to the next `if` block. Could you refactor this to remove the duplication (e.g. extract the commonality into a new method(s))?

**Comment by knjk04** [comment]
How are you getting on with the changes requested?

**Comment by tjrelyts** [comment]
I'm very sorry for my lack of responses and timing. Things have gotten a bit busy. I've been trying to figure out Docker, as I am still unable to run the program. That is really the only thing holding me back at the moment. I am working on getting it fixed now, however. My apologies.

**Comment by knjk04** [comment]
No, that's fine. I'm glad you got Docker working.

**Comment by knjk04** [comment]
How are you getting on?

**Comment by tjrelyts** [comment]
Hello! My computer had multiple issues recently with crashes and corrupted files that resulted in me resetting my PC. I'm sorry I haven't really been able to make any progress as of the moment. Would it be possible to unassign me and assign someone else, as I am currently getting mroe busy and having some issues with my computer?

**Comment by knjk04** [comment]
That's fine
